### ISSUE TITLE
:verified: emoji in profile box with website

### ISSUE THREAD

**Issue by eloydegen** [mastodon/mastodon#20316]
### Pitch

Currently Mastodon has a website verification feature. It has no account verification, but as a joke it still implemented a :verified: emoji. I like it! However, it is also possible to use this emoji in a profile box that has a URL. This collides with a real verification feature and is potentially misleading. Not all users know that this field should be green. Example:

![image](https://user-images.githubusercontent.com/3866292/201074837-5332d469-5f8d-454e-b691-477e9920e8e8.png)

Should Mastodon disallow this emoji in a field that contains a URL?

### Motivation

Make sure end users are not being misled.

**Comment by eloydegen** [comment]
Ah, it's not even upstream, but a customization done by multiple instances. My bad!

**Comment by ineffyble** [comment]
Oh, that's an annoying one. Perhaps validation should prevent all emoji if there's a URL in there. Hmm. @Gargron any ideas?

**Comment by timvisee** [comment]
This could be solved by linking an URL XOR showing emoji's. Meaning: either link an URL or show emoji's if there's none, not both together.

**Comment by eloydegen** [comment]
@timvisee Yeah, that would work, but it would also make other fun and non misleading emojis impossible to use, which I'd ideally like to avoid. But since other instances could make a :verified_hacked: emoji with the same content (that gets copied to other instances as an image), I think that's the only truly scalable solution for this problem.

**Comment by ineffyble** [comment]
I can't replicate this, in that I can't get a custom emoji to render in the metadata field on either Mastodon 3.5.3 or 4.0.0rc2. Is there something I'm missing?

**Comment by ineffyble** [comment]
> This could be solved by linking an URL XOR showing emoji's. Meaning: either link an URL or show emoji's if there's none, not both together.

Even this might not be enough, because a domain name that isn't a URL/link isn't that different visually from one that was.

<img width="850" alt="image" src="https://user-images.githubusercontent.com/834880/201090934-4388a471-422d-4f7b-a59d-a4e08d7c3666.png">

**Comment by eloydegen** [comment]
@ineffyble Well, that would at least prevent people from clicking on a malicious website since they have to copy it.

**Comment by saschanaz** [comment]
There's a Mastodon server abusing this at instance-level. https://fosstodon.org/@krosylight/109373942627329275

Obviously we can't prevent customized instances from allowing the check marks. So the process could be:

1. Disallow new metadata field from having emojis and URL together (as a default local behavior)
2. Strip or replace emojis when showing existing profile data at API level

This way we don't need to touch the database.
